### PR TITLE
Tighten dimensions

### LINKED ISSUE #11: Tighten dimensions restrictions
Following existing restrictions on `pixelHeight` and `pixelDepth`, the spec should state that `pixelWidth` cannot be `0`.

Common dimensions combinations could be organized like this:

| Type | `pixelWidth` | `pixelHeight` | `pixelDepth` | `numOfArrayElements` | `numOfFaces` |
|------------|:------------:|:-------------:|:------------:|:-----------------------:|:---------------:|
| 1D | > 0 | 0 | 0 | 0 | 1 |
| 2D | > 0 | > 0 | 0 | 0 | 1 |
| 3D | > 0 | > 0 | > 0 | 0 | 1 |
| CUBE | > 0 | > 0 | 0 | 0 | 6 |
| 1D_ARRAY | > 0 | 0 | 0 | > 0 | 1 |
| 2D_ARRAY | > 0 | > 0 | 0 | > 0 | 1 |
| CUBE_ARRAY | > 0 | > 0 | 0 | > 0 | 6 |

Any combination that doesn't fit the table above should be declared invalid and the loader should refuse to continue reading a file.

We should decide whether to allow (given that there's no API support for these):
- incomplete cubemaps (`numOfFaces > 1 && numOfFaces < 6`);
- 3D arrays (`pixelDepth > 0 && numOfArrayElements > 0`);
- cubemaps with 3D arrays (`pixelDepth > 0 && numOfArrayElements > 0 && numOfFaces == 6`).

### PR DESCRIPTION
**Pull request by lexaknyazev**
Fixes #11.

### CODE REVIEW

File: ktxspec.adoc [hunk 1]
```diff
@@ -283,23 +283,33 @@ https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#features-form
...
 === numberOfFaces
 `numberOfFaces` specifies the number of cubemap faces. For cubemaps
-and cubemap arrays this should be 6. For non cubemaps this should
-be 1. Cube map faces are stored in the order: +X, -X, +Y, -Y, +Z,
--Z.
+and cubemap arrays this must be 6. For non cubemaps this must be 1.
+Cube map faces are stored in the order: +X, -X, +Y, -Y, +Z, -Z.
+
+Applications willing to store incomplete cubemaps should flatten faces
+into 2D array and use <<KTXcubemapIncomplete>> metadata to signal which
```
Comment (MarkCallow) [inline ktxspec.adoc]:
```suggestion
into a 2D array and use <<KTXcubemapIncomplete>> metadata to signal which
```

File: ktxspec.adoc [hunk 2]
```diff
@@ -623,6 +633,23 @@ supplemented by extensions. The references are given as examples
 and do not imply that all of those texture types can be loaded in
 any particular version of OpenGL {,ES} or Vulkan.
 
+=== Texture Type
+The type of texture can be determined from the following table. Any 
+other combination of parameters makes KTX file invalid.
```
Comment (MarkCallow) [inline ktxspec.adoc]:
```suggestion
other combination of parameters makes the KTX file invalid.
```

File: ktxspec.adoc [hunk 3]
```diff
@@ -727,6 +754,39 @@ Only the most recent writer should be identified.  Editing tools
 should overwrite this value when rewriting a file originally written
 by a different tool.
 
+=== Incomplete Cubemap
+KTX file can be used to store an incomplete cubemap or an array of
```
Comment (MarkCallow) [inline ktxspec.adoc]:
```suggestion
A KTX file can be used to store an incomplete cubemap or an array of 
```

File: ktxspec.adoc [hunk 4]
```diff
@@ -727,6 +754,39 @@ Only the most recent writer should be identified.  Editing tools
 should overwrite this value when rewriting a file originally written
 by a different tool.
 
+=== Incomplete Cubemap
+KTX file can be used to store an incomplete cubemap or an array of 
+incomplete cubemaps. In such case, <<numberOfFaces>> must be `1` and
```
Comment (MarkCallow) [inline ktxspec.adoc]:
```suggestion
incomplete cubemaps. In such a case, <<numberOfFaces>> must be `1` and 
```

**Comment by lexaknyazev** [comment]
@MarkCallow 
Looks like all language changes above haven't been applied on merge. Should I open a new PR?

**Comment by MarkCallow** [comment]
Yes please. It looks like the new suggested change feature didn't work as intended or I misunderstood it. I assume I was the PR "assignee". I guess I should have done "Resolve conversation" first. Oops!

Make sure to pull the latest from upstream. I just spotted a fixed an unrelated typo.

**Pull request merged**